### PR TITLE
Add sphinx support for documentation

### PR DESCRIPTION
**Pull request by dzier**
There are a lot of new files here needed for Sphinx. For review, please look at changes made to .md files, or any new .md files. DM me for a link to the compiled documentation version to see what it will look like.

### CODE REVIEW

File: docs/contents.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+:caption: Getting Started
+
+getting_started/quickstart
+```
+
+```{toctree}
+:maxdepth: 1
+:caption: User Guide
+
+user_guide/performance_tuning
```
Comment (GuanLuo) [inline docs/contents.md:40]: Is this code generated? Or we need to add the pointer to newly added markdown file to properly render the table of content.

File: docs/customization_guide/build.md [hunk 2]
```diff
@@ -274,11 +274,11 @@ issues since non-supported versions are not tested.
 ## <a name="windows"></a>Building for Windows 10
```
Comment (GuanLuo) [inline docs/customization_guide/build.md:274]: FYI that we can remove the a tag if we use title a ref link instead.

**Review by GuanLuo — DISMISSED** [review]

File: docs/contents.md (hunk 1 above)
Comment (dzier) [reply docs/contents.md:40]: It is not user generated. It is the templates that sphinx uses to render the left side nav bar. It is some sphinx magic that makes it work and is not intended to be read on GitHub. Note the `{toctree}` at the top of the section. This tells Sphinx to render a table of contents and it has it's own method to generate the correct links from this.  See https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-toctree for more info.

**Comment by tanayvarshney** [comment]
Can we add the link to the hosted docs in the about section?

File: docs/contents.md (hunk 1 above)
Comment (GuanLuo) [reply docs/contents.md:40]: I meant to ask that if we are adding new documentation page for new feature (say "protocol/extension_new_feature.md"), do we need to manually add the entry "protocol/extension_new_feature" in this file to make sure it is included in the rendered documentation.

I was hoping that this file is automatically generated by a script that scans the file directory for documentation

**Review by GuanLuo — APPROVED** [review]

**Pull request merged**